### DISCUSSION TITLE
Indicate breaking changes and bump MAJOR semver version for upcoming releases

### DISCUSSION THREAD

**Discussion by MichelML** [epam/ketcher#1660] [General]
Hi there,

I'm currently updating the ketcher from 2.0.0 (before any npm releases were available) to 2.5.0, now using ketcher-react.

I found that the two versions have some significant breaking changes, such as

1) major style changes
2) getSmiles() and other get methods now using promises instead of returning primitive values.
3) having multiple ketcher on a single page appears to be non-obvious now

I think any of these three things would have deserved a MAJOR semver bump (moving from 2.x to 3.x). All this to say, for next major releases and changes like this, a MAJOR semver version should be bumped, as well as all breaking changes listed and ways to bypass those if users want to migrate to the next major version.

Cheers and continue this awesome work, so nice to have an open source ketcher like this :pray: 

cc @kolya-ay @AndreiMazol @ElenaOdnoshivkina @AnaMoskvina

**Comment by AndreiMazol** [comment]
Dear @MichelML ,

Thank you for interest to Ketcher and sorry for inconvenience.

Changes described in point 1 did not lead to change major version because styles changes are just look and feel and affect only UI. It is possible to  fully rewrite Ketcher styles providing an additional CSS. We plan to allow to change Ketcher theme by configuring <Editor> component in the future releases.

Regarding point 2, you are probably right, but Ketcher follows roadmap witch includes changes from 2.1 without incremental major version.

If we are talking about point 3, you still can have multiple Ketcher instances by wrapping ready-to-run app into  IFrame (you  can find it in Assets block of every release). If you want to make it by using react package, please use onInit property on <Editor> to get  the link  to valid Ketcher object (please look at https://github.com/epam/ketcher/blob/6ebb07390cc680330525132f8a45279320117ef7/example/src/App.tsx#L62 as example - here we save Ketcher object as a property of global object but you can save it into your own storage to be able manage it).

Release 3.0 will include (major features):

- possibility to work with ketcher-core without any framework/library - it will be possible to execute all critical operations from ketcher API
- refactoring of ketcher-react should be done: this package should not have any business logic. (ketcher-react should be a Presentation layer of core module).
- possibility to  create Presentation layer for any popular library/framework  in the shortest time period.

Best regards,
Ketcher team
